### DISCUSSION TITLE
Memory Leak?

### DISCUSSION THREAD

**Discussion by beiarn** [ScratMan/HASmartThermostat#69] [General]
Other ppl having memory_leak issues with this custom_integration?

leak was continuing and home assistant running unstable and rebooting after a while.

I just disabled the integration and everything went bak to normal.

I will reinstall later tonight and se if it's the autotuner thats the issue.

I really like this integration, so it would be nice if others having the same issue come forward. 
That will give ScratMan more input and maybe he will look into it.

EDIT: Looks like the leak is connected to the autotune function.

![image](https://user-images.githubusercontent.com/93034084/149636539-8c09e0a2-0065-433b-988c-9129eb515c24.png)

**Comment by ScratMan** [comment]
The Autotune function stores the temperature sensor data in memory for the full lookback duration specified in the configuration, and depending on the sampling rate of the sensor, this may be huge.

**Reply by beiarn** [reply]
Hi.
Thanks for your response and for your great "app". :)

Will try to reduce the lookback and temperature on floors next to the room i'm trying to find PID values for.

I have to say that I have used the autotune settings before. 
Only difference when autotuning this floor, is that this room has the most stable floor temperature for all om my floors(center of the house).

**Reply by beiarn** [reply]
Hi again.
This is the memory usage while autotuning with lower lookback settings. 
No memory leak this time or did not run out of memory. hmm. 

 Used settings earlier:
#- platform: smart_thermostat
#  unique_id: smart_thermostat_vk_vaskerom
#  name: S-VK Vaskerom
#  heater: switch.vk_vaskerom
#  target_sensor: sensor.vk_vaskerom_air_temperature
#  min_temp: 16
#  max_temp: 23
#  ac_mode: False
#  target_temp: 21.0
#  keep_alive:
#    seconds: 60
#  sampling_period: 00:05:00
#  target_temp_step: 0.1
#  min_cycle_duration: 00:02:00
#  kp : 34
#  ki : 40
#  kd : 160
#  pwm : 00:05:00
#autotune: "ziegler-nichols"
#noiseband: 0.1
#lookback: 10:00:00 (success with 05:00:00)

![image](https://user-images.githubusercontent.com/93034084/149984691-6deec33f-60b5-40f4-95ad-4eb93d5a8106.png)
